File: app/models/line.py
from odmantic import Model, Reference
from datetime import time
from typing import List
from app.models.stop import Stop
from app.models.category import Category

class Line(Model):
    name: str
    category: Category = Reference()
    created_at: str  # format YYYY-MM-DD
    start_time: time
    end_time: time
    stops: List[Stop] = []  # ordonnés

File: app/models/line.js
const mongoose = require("mongoose");

const lineSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true
  },
  createdAt: {
    type: String, // format YYYY-MM-DD
    required: true
  },
  startTime: {
    type: String, // format HH:MM
    required: true
  },
  endTime: {
    type: String, // format HH:MM
    required: true
  },
  category: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Category",
    required: true
  },
  stops: [
    {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Stop"
    }
  ]
});

module.exports = mongoose.model("Line", lineSchema);
